### PR TITLE
Support boot/vendorboot image v4 and bootconfig feature

### PR DESCRIPTION
**Pull request by ceiba1985**
Android 12 introduced boot/vendorboot image v4 and bootconfig
feature. Basing on the vts requirement, cmdline parameters
prefixed with 'androidboot' should be moved to bootconfig.

Tracked-On: OAM-100125
Signed-off-by: ji, zhenlong z <zhenlong.z.ji@intel.com>

### CODE REVIEW

File: libkernelflinger/android.c [hunk 1]
```diff
@@ -1169,9 +1295,11 @@ static EFI_STATUS setup_command_line(
         if (EFI_ERROR(ret))
                 goto out;
 
-        ret = add_bootvars(bootimage, &cmdline16);
-        if (EFI_ERROR(ret))
-                goto out;
+        if (aosp_header->header_version < BOOT_HEADER_V3) {
```
Comment (GangSecurity) [inline libkernelflinger/android.c:1298]: any reason to add this check?

File: libkernelflinger/android_vb2.c [hunk 2]
```diff
@@ -295,7 +295,7 @@ EFI_STATUS android_install_acpi_table_avb(AvbSlotVerifyData *slot_data)
         android_query_image_from_avb_result(slot_data, "boot", &image);
         if (image != NULL) {
                 hdr = (struct boot_img_hdr *)image;
-                if ((hdr->header_version > 1) && (hdr->acpi_size > 0)) {
```
Comment (GangSecurity) [inline libkernelflinger/android_vb2.c:298]: why change >1 to ==2?

File: libkernelflinger/security.c [hunk 3]
```diff
@@ -133,7 +133,7 @@ EFI_STATUS update_rot_data(IN VOID *bootimage, IN UINT8 boot_state,
         rot_data.verifiedBootState = boot_state;
 
         temp_version.value = boot_image_header->os_version;
-        if (boot_image_header->header_version == BOOT_HEADER_V3) {
+        if (boot_image_header->header_version >= BOOT_HEADER_V3) {
                 struct boot_img_hdr_v3 *boot_hdr = (struct boot_img_hdr_v3 *)bootimage;
```
Comment (GangSecurity) [inline libkernelflinger/security.c:137]: If boot_image_header->header_version == BOOT_HEADER_V4, can struct boot_img_hdr_v3 still be used here?

File: libkernelflinger/android.c (hunk 1 above)
Comment (ceiba1985) [reply libkernelflinger/android.c:1298]: bootvars are extracted from the second stage of boot image, this concept only exists in the header version below v3. It's not compatible with the header version v3 and v4.

File: libkernelflinger/android_vb2.c (hunk 2 above)
Comment (ceiba1985) [reply libkernelflinger/android_vb2.c:298]: since acpi_size only exist in the header version below v3.

File: libkernelflinger/security.c (hunk 3 above)
Comment (ceiba1985) [reply libkernelflinger/security.c:137]: Yes, boot_header_v4 inherit from boot_header_v3, and only add a new field signature_size at the end of the header struct, and signature_size is not used here.

**Review by GangSecurity — APPROVED** [review]

**Review by JeevakaPrabu — APPROVED** [review]

**Review by kwang13 — APPROVED** [review]

**Pull request merged**